### PR TITLE
Replace OCP4 deployinprogress file with wait for process

### PR DESCRIPTION
**Pull request by jkupferer**
##### SUMMARY

Improvement for detecting OpenShift installation completion for OCP4 installer.

This implements waiting for the `openshift-installer` process to disappear rather than relying on a lockfile. This should fix issues seen in Ansible Tower where a canceled job does not run `block` `always` sections.

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

ocp4-ha-lab config
ocp4-shared config
ocp4-workshop config
host-ocp4-destroy role

### CODE REVIEW

**Comment by wkulhanek** [comment]
We may need this also in `ocp4-cluster` config.

File: ansible/configs/ocp4-ha-lab/destroy_env.yml [hunk 1]
```diff
@@ -170,17 +170,20 @@
...
 
-        - stat:
+        - name: Wait for openshift-install to complete
+          command: pidof openshift-install
+          register: r_check_openshift_install_running
+          failed_when: r_check_openshift_install_running.rc == 0
+          changed_when: false
+          until: r_check_openshift_install_running.failed
+          retries: 360
+          delay: 10
```
[the remark this answers is not in the document]
Comment (jkupferer) [reply ansible/configs/ocp4-ha-lab/destroy_env.yml:180]: Did you check how it behaves when retries is exhausted?

For example:

```
---
- hosts: localhost
  gather_facts: false
  connection: local
  tasks:
  - name: Example
    command: pidof systemd
    register: r_pidof
    failed_when: false
    until: r_pidof.rc != 0
    retries: 1

  - debug: var=r_pidof
```

[the remark this answers is not in the document]
Comment (fridim) [reply ansible/configs/ocp4-ha-lab/destroy_env.yml:180]: Good point.
OK!

**Comment by fridim** [comment]
I'm wondering now..

In case the destroy is executed during provision, instead of waiting, what about this:

- check if openshift-install is running
- if it is: kill it && continue

Would that work ?
What SIG can we send to the installer so it gracefully stops ?

**Comment by fridim** [comment]
Replying to myself:

terraform doesn't like to be killed or stopped and it's risky as it can compromise the destroy.

**Pull request merged**